flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/connector/datagen/table/RandomGeneratorVisitor.java

```diff
@@ -135,7 +135,7 @@ public DataGeneratorContainer visit(CharType charType) {
                                 + "."
                                 + DataGenConnectorOptionsUtil.LENGTH)
                         .intType()
-                        .defaultValue(RANDOM_STRING_LENGTH_DEFAULT);
+                        .defaultValue(charType.getLength());
```

Comment:
After the dev reaches a concensus that the behavior change is accepted, please adapt the doc as well.

Reply:
done :)